src/libertem/executor/utils/dask_buffer.py

```diff
@@ -0,0 +1,156 @@
+import dask.array as da
+
+from ...common.math import prod
+from ...common.slice import Slice
+from ...common.buffers import BufferWrapper
+from .dask_inplace import DaskInplaceWrapper
+
+
+class DaskBufferWrapper(BufferWrapper):
```

Comment:
Someting I missed in my first review: the `BufferWrapper` supports `__array__`, so you can use `np.array(res['intensity'])` or similar code to get the result as a `numpy` array. This could be an avenue for writing code that is compatible with both `DelayedJobExecutor` and the others. Currently, this gives a "ValueError: object __array__ method not producing an array" - I think something like this could work:

```python
def __array__(self):
    return np.array(self.data)
```

Speaking of compatibility, and sorry this remark comes so late in the process: have you considered, instead of returning the dask array in `.data`, to keep `.data` eager, and adding a `.delayed` attribute instead? As far as I can see, this would mean results from the `DelayedJobExecutor` would be fully compatible with other executors, and only add the additional functionality as additional API. Is there anything in the grand design of things that speaks against this, other than having to rewrite a bunch of tests and examples (which I volunteer to help with)?

Reply:
I also had a look at `dask.array`, and sadly there doesn't seem to be an equivalent `__dask_array__` method we can implement to support `da.array(res['intensity'])`-style conversion to a dask array...

Reply:
@sk1p good idea with the `.delayed`. How should we handle `.raw_data` in this case? Add a `.raw_delayed`?

Reply:
:+1: - would also be happy with more verbose but clearly symmetric names of `.data_delayed` and `.raw_data_delayed` (but that's now a bit bike shedding :))

Reply:
This is done in 9c833965 and follow-up commits